### PR TITLE
Dashboard

### LINKED ISSUE #4754: Build Dashboard Front-End
### User Story

As a funder on the Gitcoin platform, I'd like a central place to check-in to see how my bounties are doing, and know the actions I need to take next.

### Why Is this Needed

No Dashboard in Gitcoin

### Description

_This task is to set up the front end of backend work created on sprint 13._

- Use information from per-bounty api that to show applicant details shown In mockup
- Display in Contributor Icon and github name
- Display high level bounty stats up top
- Break up page in 3 sections
- Open Bounties, Submitted Bounties, Expired Bounties (see mockup)

**Bounties Section**
- Display data end points for elements that are displayed
- Display _Bounty:_ order by most recent applicants
- Display_Applicants:_ order by the time each applied
- Display pulled in data for all open bounties. 
- Allow ability to accept/reject on the front-end
- Display all bounties
- Allow clicking to user directory in case the right contributors don’t apply.

**Submission Section**
- Display according to design, _Bounty:_ Order by most recent applicant submission
- Display_Applicants:_ order by the time each contributor submitted, recent first.
- Display user name eth address, pr info, commit count
- Allow ability to accept and reject

**Expired bounty**
- Display the relevant information
- Order based on recently expired bounty
- Clicking extend expiration takes user to issue details with expiration date modal open.

### Current Behavior
No Front-End

### Definition of Done
Front-end is built as shown in design

* Depending on how much we get done in sprint 14, product team to decide when to release this feature.
* Also depends on feedback from continued dashboard research that is happening.

### Data Requirements
- Number of click actions taken on the dashboard

### See Design Here
[See the Latest design ](https://consensys.invisionapp.com/share/NYSOGHX2D8R#/371051245_Dashboard_-1-)

### PR DESCRIPTION
**Pull request by octavioamu**
##### Description

- Funder dashboard with 3 areas and actions:
OPEN BOUNTIES (applicants) 
BOUNTY SUBMISSIONS (applicants) 
EXPIRED BOUNTIES
- Contributor dashboard with 3 areas and actions:
BOUNTIES IN PROGRESS
BOUNTIES APPLIED 
WORK SUBMITTED

General notes:
"view applicants"  only request data on click the first time is clicked
Each dashboard (funder, contributors)  only request data when the tab is clicked
Both have a loading state and empty data.
The default tab is "funder" but if you have a persona "hunter" it will be contributor tab. 
 
 
##### Refers/Fixes

ref #4153
Fix #4754
##### Testing

https://embed.vidyard.com/share/JBaBMgybeaqdVnQuFqnJNn?

### CODE REVIEW

**Comment by owocki** [comment]
just pulled this down + been playing with it.  nice work, coming along very well!

some feedback.  some nitpicky.. trying to focus on 'attention to detail' 

# funder dashboard feeedback

speed / performance is wayyy better than i expected for my user (owocki) that has many bounties :)

these buttons seem misaligned http://bits.owocki.com/32a9ed53dcd2/Image%202019-07-29%20at%203.00.51%20PM.png

 i ahve 130 open bounties .  its hard to scroll past them all.  maybe set a max-height and a y-scroll on this div? 
http://bits.owocki.com/37b0c067580c/Image%202019-07-29%20at%203.01.12%20PM.png

feature/invite buttons point to http://localhost:8000/dashboard/bounty/change/4715/ which seeems to 404 for me.. 

inconsistent capitalization here and “0 applicantions” is (1) spelled wron. http://bits.owocki.com/954d2d34aae1/Image%202019-07-29%20at%203.02.10%20PM.png

i dont think that the ‘accept/reject’ buttons are correctly showing on my pending bounties because i get this screen when i click ‘accept’ http://bits.owocki.com/786cf2ae1f70/Image%202019-07-29%20at%203.03.03%20PM.png -- it should likely only show applicants who are in a pending state (not already approved)

i think that the ‘feature bounty ‘and ‘invite contributors’ buttons should show up for all 3 categories of bounties.  its possible that i’m not happy with the marketing of my bounties in all three scenarios (open bounties, submitted bounties, expired bounties)

there shoudl possibly also be a ‘contact the gitcoin core team’ button on these rows, in case the self service stuff isnt’ working.

title should be capitalized http://bits.owocki.com/512ce47c9140/Image%202019-07-29%20at%203.05.44%20PM.png

responsive css could use a little love http://bits.owocki.com/3b2dfa7e0f70/Image%202019-07-29%20at%203.06.16%20PM.png

theres a bunch of empty space up here ( http://bits.owocki.com/e0c0c20739b2/Image%202019-07-29%20at%203.06.29%20PM.png ) is that by design?

# contributor dashboard
‘stop work’ has no callback -- no indicator of success when the HTTP request responds http://bits.owocki.com/bbb4677ee3e1/Screen%20Recording%202019-07-29%20at%2003.07%20PM.gif

**Comment by octavioamu** [comment]
Thanks @owocki I work in that fixes!

**Comment by octavioamu** [comment]
@owocki all done except for: 

> i dont think that the ‘accept/reject’ buttons are correctly
I wasn't able to reproduce, I think can be related to the db? the query just do exactly that looks for open bounties . What bounty type is the one on your print?

> i think that the ‘feature bounty ‘and ‘invite contributors’ buttons should 
There is a very small space right now so for that a redesign will be needed 

> there shoudl possibly also be a ‘contact the gitcoin core team’ button 
I think this need to be solved with the zendesk or any more global solution instead of here, but anyway I let that for the design team.

**Comment by owocki** [comment]
> I wasn't able to reproduce, I think can be related to the db? the query just do exactly that looks for open bounties . What bounty type is the one on your print?

its not a bounty type, its a bounty.interests thing.  we need bounty.interests.filter(pending=True) on it

**Comment by octavioamu** [comment]
I will check it don't know what is happening with github my last commit is not here https://github.com/gitcoinco/web/pull/4588/commits/db1f71da98bdb2e69f992bc855acc66ac3699868
image

**Comment by octavioamu** [comment]
> I will check it don't know what is happening with github my last commit is not here [db1f71d](https://github.com/gitcoinco/web/commit/db1f71da98bdb2e69f992bc855acc66ac3699868)
> image

At least now the commit link is working

**Comment by octavioamu** [comment]
@owocki added the pending filter, now should be good. 
In the list there is a missing commit about mobile but I think is in git, should be just a github interface bug.

File: app/app/tests/test_app_urls.py [hunk 1]
```diff
@@ -89,14 +89,14 @@ def test_faucet_resolve(self):
...
 
-    def test_dashboard_reverse(self):
-        """Test the dashboard url and check the reverse."""
-        self.assertEqual(reverse('dashboard'), '/dashboard')
-
-    def test_dashboard_resolve(self):
-        """Test the dashboard url and check the resolution."""
-        self.assertEqual(resolve('/dashboard').view_name, 'dashboard')
-        self.assertEqual(resolve('/dashboard/').view_name, 'dashboard')
+    # def test_dashboard_reverse(self):
```
Comment (danlipert) [inline app/app/tests/test_app_urls.py]: should we just remove these?

Comment (thelostone-mc) [reply app/app/tests/test_app_urls.py]: lol

**Pull request merged**